modules/product/src/ProductVariationManager.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\commerce_product\ProductVariationManager.
+ */
+
+namespace Drupal\commerce_product;
+
+use Drupal\commerce_product\Entity\Product;
+use Drupal\commerce_product\Event\AvailableVariationsEvent;
+use Drupal\commerce_product\Event\ProductEvents;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+class ProductVariationManager implements ProductVariationManagerInterface {
+
+  /**
+   * The variation storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $variationStorage;
+
+  /**
+   * The event dispatcher.
+   *
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  protected $eventDispatcher;
+
+
+  /**
+   * Constructs a ProductVariationManager object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entityManager
+   * @param \Symfony\Component\EventDispatcher\EventDispatcher $eventDispatcher
+   */
+  public function __construct(EntityManagerInterface $entityManager, EventDispatcher $eventDispatcher) {
+    $this->variationStorage = $entityManager->getStorage('commerce_product_variation');
+    $this->eventDispatcher = $eventDispatcher;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEnabledVariations(Product $product) {
+    $variations = $product->variations;
+    $enabledVariations = [];
+
+    $query = \Drupal::entityQuery('commerce_product_variation')
+    ->condition('status', TRUE);
+    $group = $query->orConditionGroup();
+
+    foreach($variations as $variation) {
+      $id = $variation->getValue();
+      $group->condition('variation_id', $id);
```

Comment:
You can just do $query->condition('variation_id', $ids); and EQ will make it an IN condition.